Initial classes

mecha opened the pull request.

mecha committed:
Installed dependencies
 Please enter the commit message for your changes. Lines starting

mecha committed:
Ignoring irrelevant PHPS files

mecha committed:
Added PHPS config for PHPUnit

mecha committed:
Removed irrelevant PHPS config files

mecha committed:
Added `FactoryAwareTrait`

mecha committed:
Added `SubjectConfigAwareTrait`

mecha committed:
Added `AbstractBaseFactoryException`

mecha committed:
Moved func tests to correct directory

mecha committed:
Added `FactoryException`

mecha committed:
Fixed subject config setter for null arg

mecha committed:
Added `CouldNotMakeException`

mecha committed:
Added `CreateFactoryExceptionCapableTrait`

mecha committed:
Added `CreateCouldNotMakeExceptionCapableTrait`

mecha committed:
Moved func tests to correct directory

mecha committed:
Added `AbstractBaseCallbackFactory`

mecha committed:
Updated dependencies

XedinUnknown commented on composer.json:
```diff
@@ -12,13 +12,23 @@
+        "dhii/factory-interface": "0.1-alpha1 | 0.1-alpha2",
+        "dhii/callback-abstract": "0.1-alpha2",
+        "dhii/invocable-interface": "0.1-alpha1",
+        "dhii/invocable-base": "dev-task/initial-classes",
```
You can now depend on [`dhii/invocable-base@0.1-alpha1`][dhii/invocable-base@0.1-alpha1].

[dhii/invocable-base@0.1-alpha1]: https://github.com/Dhii/invocable-base/releases/tag/v0.1-alpha1

XedinUnknown commented on composer.json:
```diff
@@ -28,5 +38,16 @@
+    "repositories": [
+        {
+            "type": "vcs",
+            "url": "http://github.com/dhii/invocable-base"
```
Now published [here](https://packagist.org/packages/dhii/invocable-base).

XedinUnknown commented on composer.json:
```diff
@@ -28,5 +38,16 @@
+    ],
+    "config": {
+        "platform": {
+            "php": "5.4.0"
```
This looks extra.

mecha committed:
Updated dependencies to stable versions

- Removed repositories entry from composer.json
- Removed platform config from composer.json

mecha committed:
Updated project dependency info

mecha committed:
Fixed use of `InvocationException`

XedinUnknown commented:
@mecha, ready for merge?

mecha commented:
@XedinUnknown Let's see if it builds successfully first

XedinUnknown commented on src/SubjectConfigAwareTrait.php:
```diff
@@ -0,0 +1,68 @@
+     *
+     * @since [*next-version*]
+     *
+     * @return array|ArrayAccess|ContainerInterface|stdClass The subject factory configuration, if any.
```
Cannot be null?

mecha replied:
Can be. Nice catch.

XedinUnknown commented on src/FactoryAwareTrait.php:
```diff
@@ -0,0 +1,90 @@
+     *
+     * @since [*next-version*]
+     *
+     * @var FactoryInterface
```
Cannot be `null`?

mecha replied:
Can be. Nice catch.

mecha committed:
Added missing `null` type possibility in docs

mecha committed:
Added missing `null` type possibility in docs
